src/jit/emit.cpp

```diff
@@ -2980,13 +2982,14 @@ emitter::instrDesc  * emitter::emitNewInstrCallInd(int        argCnt,
         mode displacement, or we have some byref registers
      */
 
-    if  (!VarSetOps::IsEmpty(emitComp, GCvars)     ||   // any frame GCvars live
-         (gcRefRegsInScratch)        ||   // any register gc refs live in scratch regs
-         (byrefRegs != 0)            ||   // any register byrefs live
-         (disp < AM_DISP_MIN)        ||   // displacement too negative
-         (disp > AM_DISP_MAX)        ||   // displacement too positive
-         (argCnt > ID_MAX_SMALL_CNS) ||   // too many args
-         (argCnt < 0)                   ) // caller pops arguments
+    if  (!VarSetOps::IsEmpty(emitComp, GCvars)  ||                              // any frame GCvars live
+         (gcRefRegsInScratch)                   ||                              // any register gc refs live in scratch regs
+         (byrefRegs != 0)                       ||                              // any register byrefs live
+         (disp < AM_DISP_MIN)                   ||                              // displacement too negative
+         (disp > AM_DISP_MAX)                   ||                              // displacement too positive
+         (argCnt > ID_MAX_SMALL_CNS)            ||                              // too many args
+         (argCnt < 0)                                                           // caller pops arguments
+        FEATURE_UNIX_AMD64_STRUCT_PASSING_ONLY( || (retSize1In != EA_UNKNOWN))) // There is a second ref/byref return register.
```

Comment:
Shouldn't the condition be retSize1In is a gc-ref or byref?  What if the second 8byte is a other than a gc-type?